### DISCUSSION TITLE
tab切换时 ，反馈信息弹出框没有隐藏

### DISCUSSION THREAD

**Discussion by PlutoCA** [alibaba/formily#2397] [Q&A 2.x] [answered]
https://codesandbox.io/s/morning-cookies-3spw3

复现步骤：
添加多页，点击提交。切换tab时，错误信息弹框没有隐藏。

**Comment by janryWang** [comment, marked as answer]
无解，或者不要用popup模式

**Reply by PlutoCA** [reply]
想了两个不成熟方案哈。
是否可以不默认渲染在body下面。
或者popup模式下触发行为改成聚焦或者点击下才显示

**Reply by janryWang** [reply]
不渲染在body上，table的问题还是会有

**Reply by PlutoCA** [reply]
> 不渲染在body上，table的问题还是会有

https://github.com/alibaba/formily/blob/d6cdf71a1cf798f315be32919af567ff3f3e3fcf/packages/antd/src/form-item/index.tsx#L191

这一段判断是否可以改成 `visible={!!feedbackText && field.active} `

1. 个人感觉popup模式场景下，点击聚焦后再弹窗提示稍微友好一点。而不是点击触发错误校验后一堆弹窗弹出来。
2. 目前弹窗隐藏时（校验错误 -> 成功）有个很明显的过渡，先清空弹窗内容，弹窗再隐藏。
3. 这样也可以解决弹窗没法正常关闭的情况

**Reply by janryWang** [reply]
那就得做到鼠标滑出，立马隐藏错误消息，否则table内部滚动就会定位失效，但是这玩意太定制了，不是普适需求啊
